crates/services/txpool_v2/src/pool.rs

```diff
@@ -300,6 +324,43 @@ where
         }
     }
 
+    fn populate_saved_outputs_cache(&mut self, best_txs: &[StorageData]) {
+        self.extracted_outputs.clear();
```

Comment:
I don't see why do we need to clear it here. This place is called several times during the block production, and clearing outputs here means we can't actually use them.

It would be nice to add a test that would catch this incorrect behavior